powerauth-restful-security-javaee/src/main/java/io/getlime/security/powerauth/rest/api/jaxrs/provider/PowerAuthAuthenticationProvider.java

```diff
@@ -287,8 +288,9 @@ public PowerAuthApiAuthentication validateToken(String tokenHeader, List<PowerAu
         final PowerAuthApiAuthentication auth;
         try {
             auth = this.authenticate(powerAuthTokenAuthentication);
-        } catch (RemoteException e) {
-            throw new PowerAuthAuthenticationException("POWER_AUTH_TOKEN_SOAP_ERROR");
+        } catch (RemoteException ex) {
+            logger.warn("Remote communication failed, error: {}", ex.getMessage());
+            throw new PowerAuthAuthenticationException("POWER_AUTH_TOKEN_REMOTE_ERROR");
```

Comment:
Let's pass `ex` as a parameter to the new exception to keep the cause somewhere.